File: app/src/main/java/tabletop/velocic/com/worldforgerpgtools/GeneratorDeserializer/Generator.java
package tabletop.velocic.com.worldforgerpgtools.GeneratorDeserializer;

public class Generator
{
    public String Name;
    public int DefaultNumResultRolls = 1;
    public TableEntries[] TableEntries;
    public String AssetPath;

    public String getName()
    {
        return Name;
    }

    public int getDefaultNumResultRolls()
    {
        return DefaultNumResultRolls;
    }

    public void setAssetPath(String assetPath)
    {
        this.AssetPath = assetPath;
    }

    public String getAssetPath()
    {
        int firstSlashIndex = AssetPath.indexOf("/");
        String firstPathComponent = "";

        if (firstSlashIndex != -1) {
            firstPathComponent = AssetPath.substring(0, firstSlashIndex);
        }

        if (firstPathComponent.equals(GeneratorImporter.GENERATOR_DATA_FOLDER)) {
            return AssetPath.substring(firstSlashIndex + 1, AssetPath.length()) + "/";
        }

        return AssetPath + "/";
    }

    public TableEntries[] getTable()
    {
        return TableEntries;
    }
}

File: app/src/main/java/tabletop/velocic/com/worldforgerpgtools/GeneratorDeserializer/Generator.kt
package tabletop.velocic.com.worldforgerpgtools.GeneratorDeserializer

import com.google.gson.annotations.SerializedName

class Generator(
    @SerializedName("Name")
    var name: String,

    @SerializedName("DefaultNumResultRolls")
    var defaultNumResultRolls: Int = 1,

    @SerializedName("TableEntries")
    var table: Array<TableEntries>,

    assetPath: String
) {
    @SerializedName("AssetPath")
    var assetPath = assetPath
        get() {
            val firstSlashIndex = field.indexOf("/")
            var firstPathComponent = ""

            if (firstSlashIndex != -1) {
                firstPathComponent = field.substring(0, firstSlashIndex)
            }

            if (firstPathComponent == GeneratorImporter.GENERATOR_DATA_FOLDER) {
                return "${field.substring(firstSlashIndex + 1, field.length)}/"
            }

            return "$field/"
        }
}